### PR TITLE
Plotting

### PR DESCRIPTION
**Pull request by YavorLalev**
A plot of the data is generated.

[US4-Plotting](https://github.com/users/catdieval/projects/1/views/1?pane=issue&itemId=55229085)

### CODE REVIEW

File: pages/_app.js [hunk 1]
```diff
@@ -1,12 +1,26 @@
...
 
 export default function App({ Component, pageProps }) {
+  const [clickedChartType, setClickedChartType] = useState("");
+
+  function handleChartType(element) {
+    setClickedChartType(element.target.innerText);
+    alert(`You chose ${element.target.innerText}`);
+  }
+
+  console.log(`At the beginning we have ${clickedChartType}`);
```
Comment (MerleMak) [inline pages/_app.js:13]:
```suggestion
```

File: Components/ChartItem/index.js [hunk 2]
```diff
@@ -1,19 +1,8 @@
...
-
-  function handleChartType(element) {
-    setClickedChartType(element.target.innerText);
-    alert(`You chose ${element.target.innerText}`);
-  }
-
+export default function ChartItem({ name, icon, handleChartType }) {
   return (
     <FlexContainer $direction="column">
       <Button onClick={handleChartType}>{name}</Button>
```
Comment (MerleMak) [inline Components/ChartItem/index.js:8]:
```suggestion
export default function ChartItem({ name, icon, onSelectChartType }) {
  return (
    <FlexContainer $direction="column">
      <Button onClick={onSelectChartType}>{name}</Button>
```

File: Components/ListOfCharts/index.js [hunk 3]
```diff
@@ -3,14 +3,19 @@ import { FlexContainer } from "../FlexContainer";
 import ChartItem from "../ChartItem";
 import { chartArray } from "../../lib/listOfPlotTypes";
 
-export default function ListOfCharts() {
+export default function ListOfCharts({ handleChartType }) {
```
Comment (MerleMak) [inline Components/ListOfCharts/index.js:6]:
```suggestion
export default function ListOfCharts({ onSelectedChartType }) {
```

File: Components/ListOfCharts/index.js [hunk 4]
```diff
@@ -3,14 +3,19 @@ import { FlexContainer } from "../FlexContainer";
...
-      <FlexContainer $direction="row">
+      <FlexContainer $grid="grid">
         {chartArray.map((chart) => {
           return (
-            <ChartItem key={chart.name} icon={chart.icon} name={chart.name} />
+            <ChartItem
+              key={chart.name}
+              icon={chart.icon}
+              name={chart.name}
+              handleChartType={handleChartType}
```
Comment (MerleMak) [inline Components/ListOfCharts/index.js:17]:
```suggestion
              onSelectChartType={onSelectChartType}
```

File: pages/index.js [hunk 5]
```diff
@@ -1,11 +1,13 @@
...
 import ListOfCharts from "../Components/ListOfCharts";
+import Plotting from "../Components/Plotting";
 
-export default function HomePage() {
+export default function HomePage({ clickedChartType, handleChartType }) {
   return (
     <div>
       <UploadData />
-      <ListOfCharts />
+      <ListOfCharts handleChartType={handleChartType} />
```
Comment (MerleMak) [inline pages/index.js:9]:
```suggestion
      <ListOfCharts onSelectChartType={onSelectChartType} />
```

File: pages/_app.js [hunk 6]
```diff
@@ -1,12 +1,26 @@
...
+
   return (
     <>
       <Layout>
         <GlobalStyle />
-        <Component {...pageProps} />
+        <Component
+          {...pageProps}
+          clickedChartType={clickedChartType}
+          handleChartType={handleChartType}
```
Comment (MerleMak) [inline pages/_app.js:22]:
```suggestion
          onSelectChartType={handleChartType}
```

File: Components/Plotting/index.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+                  ticks: "outside",
+                },
+                width: 600,
+                height: 500,
+              }}
+              config={{
+                displayModeBar: true,
+                modeBarButtonsToRemove: ["lasso2d", "select2d", "pan2d"],
+              }}
+            ></Plot>
```
Comment (MerleMak) [inline Components/Plotting/index.js:52]:
```suggestion
            />
```

**Review by MerleMak — COMMENTED** [review]
Great job!
Just some naming conventions to improve :)
 
Keep it up 🚀 Really great direction guys!

File: pages/_app.js (hunk 1 above)
Comment (YavorLalev) [reply pages/_app.js:13]: Done.

File: Components/ChartItem/index.js (hunk 2 above)
Comment (YavorLalev) [reply Components/ChartItem/index.js:8]: Done.

File: Components/ListOfCharts/index.js (hunk 3 above)
Comment (YavorLalev) [reply Components/ListOfCharts/index.js:6]: Done.

File: Components/ListOfCharts/index.js (hunk 4 above)
Comment (YavorLalev) [reply Components/ListOfCharts/index.js:17]: Done.

File: pages/index.js (hunk 5 above)
Comment (YavorLalev) [reply pages/index.js:9]: Done.

File: pages/_app.js (hunk 6 above)
Comment (YavorLalev) [reply pages/_app.js:22]: Done.

File: Components/Plotting/index.js (hunk 7 above)
Comment (YavorLalev) [reply Components/Plotting/index.js:52]: Done.

File: Components/ListOfCharts/index.js [hunk 8]
```diff
@@ -3,14 +3,19 @@ import { FlexContainer } from "../FlexContainer";
 import ChartItem from "../ChartItem";
 import { chartArray } from "../../lib/listOfPlotTypes";
 
-export default function ListOfCharts() {
+export default function ListOfCharts({ onSelectChartType }) {
   return (
     <>
       <Heading>Step 2: Select the type of chart</Heading>
-      <FlexContainer $direction="row">
+      <FlexContainer $grid="grid">
```
Comment (MerleMak) [inline Components/ListOfCharts/index.js:10]:
```suggestion
      <Container $grid="grid">
```

Flex and Grid are two completely different layout tools, either you name the styled component "Container" or you create a GridContainer and a FlexContainer as two separate things :)

**Review by MerleMak — COMMENTED** [review]
Very nice guys! Just one last comment :))

File: Components/Container/index.js [hunk 9]
```diff
@@ -0,0 +1,19 @@
+import styled, { css } from "styled-components";
+
+export const Container = styled.section`
+  flex-direction: ${(props) =>
+    props.$direction === "column" ? "column" : "row"};
+  display: flex;
+  flex-wrap: wrap;
+  justify-content: space-around;
```
Comment (MerleMak) [inline Components/Container/index.js:8]:
```suggestion
 ${(props) =>
    props.$flex === "flex" &&
    css`
      display: flex;
      flex-wrap: wrap;
      justify-content: space-around;
      flex-direction: column
    `};
```

Do you need the flex container row at this point?

Comment (MerleMak) [reply Components/Container/index.js:8]: After correcting this you can send the PR to QA :))

File: Components/ListOfCharts/index.js (hunk 8 above)
Comment (YavorLalev) [reply Components/ListOfCharts/index.js]: Done.

**Pull request merged**